early-return/src/workflows.ts

```diff
@@ -0,0 +1,15 @@
+import * as wf from '@temporalio/workflow';
+import { getTransactionConfirmation, TransactionReport } from './shared';
+
+export async function transactionWorkflow(): Promise<TransactionReport> {
+  let confirmed = false;
+  wf.setHandler(getTransactionConfirmation, async () => {
+    await wf.condition(() => confirmed);
+    const status = 'confirmed' as const;
+    return { status };
+  });
+  await wf.sleep(1000);
```

Comment:
As it is now, both the Update and the Workflow completes in the same WFT. You should have a sleep _after_ `confirmed = true` (e.g. before completing the Workflow). I have no opinion though on either it is pertinent to have a `sleep` _before_ confirming the update.

Reply:
Also, I would make the sleep duration between update completion and wf completion to be something significant… Let's say 5, or even 10 seconds. And add a comment on that sleep saying would be be some long operation that can be done asynchrnously…

Reply:
Good call, done.